customers/models.py

```diff
@@ -35,6 +38,186 @@ def get_queryset(self):
+                        postal_code=organization.get("postal_code"),
+                        city=organization.get("city"),
+                        business_id="-"
+                        if organization_type == OrganizationType.COMPANY
+                        else "",
+                    )
+
+                boats = []
+                for boat in customer_data.get("boats", []):
+                    boat_type = BoatType.objects.language("fi").get(
+                        translations__name=boat.get("boat_type")
+                    )
+                    boats.append(
+                        Boat.objects.create(
+                            owner=customer,
+                            name=boat.get("name"),
+                            boat_type=boat_type,
+                            registration_number=boat.get("registration_number"),
+                            width=Decimal(boat.get("width")),
+                            length=Decimal(boat.get("length")),
+                            draught=Decimal(boat.get("draught"))
+                            if boat.get("draught")
+                            else None,
+                            weight=Decimal(boat.get("weight"))
+                            if boat.get("weight")
+                            else None,
+                        )
+                    )
+
+                leases = []
+                for lease in customer_data.get("leases", []):
+                    berth = Berth.objects.get(
+                        pier__harbor=Harbor.objects.get(
+                            servicemap_id=lease.get("harbor_servicemap_id")
+                        ),
+                        pier__identifier=lease.get("pier_id", "-"),
+                        number=lease.get("berth_number"),
+                    )
+                    leases.append(
+                        BerthLease.objects.create(
+                            customer=customer,
+                            berth=berth,
+                            boat=boats[lease.get("boat_index")]
+                            if lease.get("boat_index")
+                            else None,
+                            start_date=lease.get("start_date"),
+                            end_date=lease.get("end_date"),
+                        )
+                    )
+
+                for order in customer_data.get("orders", []):
+                    lease = None
+                    if berth_data := order.get("berth"):  # noqa
+                        berth = Berth.objects.get(
+                            pier__harbor=Harbor.objects.get(
+                                servicemap_id=berth_data.get("harbor_id")
+                            ),
+                            pier__identifier=berth_data.get("pier_id"),
+                            number=berth_data.get("berth_number"),
+                        )
+
+                        start_date = parse(order.get("created_at")).date()
```

Comment:
Would it make sense to extract this whole start/end date logic to own module/util and unit test it?

Reply:
Hmmm. As I see it,
Pros:
* Remove complexity from the import function
* Easier to test

Cons:
* I don't see any other use case (at least not yet) which would benefit from having it as a separate util

So I guess pros outweight the cons?

Reply:
Good analysis. I think pros outweight the cons, and I personally prefer pretty small focused functions.

Reply:
Looks very good now, and great test.